packages/amplify_datastore/lib/amplify_datastore.dart

```diff
@@ -61,6 +78,27 @@ class AmplifyDataStore extends DataStorePluginInterface {
     return _instance.configureModelProvider(modelProvider: modelProvider);
   }
 
+  @override
+  Future<void> configureDataStore(
+      {ModelProviderInterface? modelProvider,
+      int? syncInterval,
+      int? syncMaxRecords,
+      int? syncPageSize}) async {
+    ModelProviderInterface provider =
+        modelProvider == null ? this.modelProvider : modelProvider;
```

Comment:
When making the changes in the interface for `modelProvider` to be nullable you will probably come across some issues here:
```suggestion
    ModelProviderInterface provider =
        modelProvider ?? this.modelProvider!;
```
Since the only usage of the `tokenOnly` constructor is internal, we should be confident in saying that the only way to construct `AmplifyDataStore` is via the default constructor where we've made the modelProvider required, so it should be safe to cast `this.modelProvider` to non-null.

Reply:
Sounds good added that change in